### ISSUE TITLE
Backup priority

### ISSUE THREAD

**Issue by cache-sk** [vmware-tanzu/velero#6310]
**Describe the problem/challenge you have**

We have many schedules scheduled at same time, which runs more than 10 minutes, and the request is that the manually triggered "velero create backup", launched even one second after that scheduled time, should be performed as first possible, sooner than backups created by schedules.

Is there a way to prioritize backup create, or at least prioritize backup create before scheduled backups?

**Describe the solution you'd like**

Some flag, which will prioritize backup create to be processed before other backups (without this flag) in queue.

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "The project would be better with this feature added"
- :-1: for "This feature will not enhance the project in a meaningful way"

**Comment by sseago** [comment]
This makes some sense, although it would need to be configurable -- i.e. for some people, scheduled backups may take priority. Also note that in the specific example  you gave, a high priority manual backup created one second after a 10-minute backup started, no matter how high the priority, it wouldn't run for 10 minutes, since only one backup is processed at once. However, if several backups were created right now from schedule and then a high priority one were created, it might be worth seeing whether it could be run after the current backup, before the others (so 10 minutes later rather than 50). That being said, this still may be problematic with the current controller design, since the already-created backups are queued up for the reconciler, and velero will reconcile these backups in the order they're queued. It would probably take a completely different controller model to handle this scenario.

It might make more sense to prioritize multi-threaded backup processing instead, since that's already a desired enhancement for other use cases -- ie. allowing more than one backup to run at the same time -- this might allow for designating specific workers for backups from schedule vs. backups created manually.

**Comment by cache-sk** [comment]
Yes, I understand that currently processed backup, will end first and will not be interrupted.

To make it clear, what i need, I'll quote myself from slack :)

As far as i know, velero is processing backups in queue, so if schedulers tiggers for example 100 backups, all will be processed gradually and if i create backup manually after "kick" time of that 100 schedulers, it will be processed as 101-th. We need to achieve somehow, that it will be processed as first in this queue, not last.
Let's say that the customer is not happy that if the scheduled backups would last an hour, then the manually created one will be performed only after an hour. He wants it immediately - for him, manually requested backup is taken as a priority.

I fully understand that some may take it completely the opposite way. They would certainly also like to have some form of prioritization for backups created by schedulers.

**Comment by danfengliu** [comment]
@cache-sk  From this opened [issue](https://github.com/vmware-tanzu/velero/issues/3554), a schedule should not trigger new backups until on-going backup by this scheduler complete, I think if this logic is accepted and implemented, and with schedule pause CLI, you can pause/unpause a schedule, then create a desired backup manually, anyway if there is an on-going backup from schedule, manually backup will be picked up after that backup.

```
 # Pause schedules named "schedule-1" and "schedule-2".
  velero schedule pause schedule-1 schedule-2

  # Unpause all schedules labeled with "foo=bar".
  velero schedule unpause --selector foo=bar
```

**Comment by sseago** [comment]
@danfengliu Although if you had 100 different schedules, all of which scheduled now, the "not trigger new backups until on-goig backup" case won't apply. The pause/unpause would certainly help if you paused all 100 of those schedules *before* they triggered their backups, though.

**Comment by danfengliu** [comment]
@sseago Got your point. Then I was thinking what if having  a `schedule pause all ` cli after fix of [issue](https://github.com/vmware-tanzu/velero/issues/3554) .

**Comment by elvisZije** [comment]
Hi guys, do you have some news about this issue ?
